### PR TITLE
Tbolt/3403 state certification letters

### LINKED ISSUE #3403: Create endpoint for uploading state certification letters
Create an endpoint that can accept the upload of a file (state certification letter) and return the URL of that file in the response.

### This task is done when...

- [x] a file can be uploaded by a user with the 'edit-state-certification' permission
- [x] file is stored in S3 or local storage (if a dev environment)
- [x] The file is checked for .docx or .pdf file type and rejected if it is a different type
- [x] Uploading a file returns a URL that can be accessed by a user with the "view-state-certification" permission

---

Add additional labels (design, dev, compliance, BUG, etc) and size before submitting.

If the issue is needed to complete prioritized work for the CURRENT SPRINT, add it to the "This Sprint" pipeline. Otherwise, all other issues will be automatically added to the _unprioritized pipeline_ for prioritization in backlog refinement or sprint planning with Product.

### LINKED ISSUE #3404: [Feature] Update the way Activities names are display to be consistent
Design ticket #3148

**Description and related issues -**
Untitled Activities show up inconsistently across the builder, review view, and PDF view and across different sections of the application. We need to update where it is display to be consistent. Use the content "Activity X: Untitled" as the pattern for what should be shown when an Activity has not been named. With X being the activity number when an Activity name needs to be populated, but does not exist.

**Mockups or link to Figma**
This will match what is currently shown in the Activities List/Dashboard

For the activity list, it should just show Untitled and no | unless there is a funding source set for the activity.
Existing Example (#2 should just be 2. Untitled):

### This task is done when…

The unnamed Activity display has been updated in
- [ ] Builder view > Activities Dashboard/List
- [ ] Export view > Activities Dashboard/List
- [ ] Builder view > Activity Overview (2X)
- [ ] Export view > Activity Overview (2X)
- [ ] Builder view > Activity Schedule Summary
- [ ] Export view > Activity Schedule Summary
- [ ] Builder view > Milestones Table
- [ ] Export view > Milestones Table
- [ ] Builder view > Proposed budget > activity breakdown
- [ ] Export view > Proposed budget > activity breakdown
- [ ] Builder view > Executive Summary
- [ ] Export view > Executive Summary
- [ ] Builder view > Headers for each page/section
- [ ] Export view > Headers for each page/section
- [ ] Builder view > Side Nav
- [ ] Builder view > Prev/Cont (Back/Next) text below the button

### PR DESCRIPTION
**Pull request by tbolt**
Resolves #3403 

**Description-**
This PR adds two new activities and a new `/auth/certifications/files` endpoint (both `GET` and `POST`). The new activities are `view-state-certifications` and `edit-state-certifications`. These are both exclusively assigned to the eAPD Federal Admin role. 

**Steps to manually verify this change...**

1. Checkout branch
2. Extract a valid **non-**eAPD Federal Admin (such as norole) jwt
3. Use an HTTP client (such as Postman/Insomnia) to hit `/auth/certifications/files` with the appropriate payload (use the OpenAPI docs to see what is valid). Test both `GET` and `POST` 
4. Verify that the response is 403 Forbidden
5. Now extract a valid eAPD Federal Admin jwt
6. Verify that you are able to successfully `POST` and `GET` the file
7. Test by using an invalid file type

### This pull request is ready to review when...

- [x] Automated tests are updated (and all tests are passing)
- [x] The change has been documented
- [x] Associated OpenAPI documentation has been updated
- [x] Changelog is updated as appropriate
- [x] The experience passes a basic manual accessibility audit (keyboard nav, screenreader, text scaling) OR an exemption is documented

### This pull request can be merged when…

- [ ] Code has been reviewed by someone other than the original author
- [ ] QA has verified the accessibility and functionality related to the change
- [ ] Design has approved the experience
- [ ] Product has approved the experience

### CODE REVIEW

**Review by knollfear — APPROVED** [review]
One small change to the test and this is GTG.

File: api/routes/auth/certifications/files.test.js [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+      );
+    });      
+  });    
+  
+  endpointTest.test('GET endpoint for fetching a state certification letter/file', async tests => {
+    let handler; 
+    
+    tests.beforeEach(async () => {
+      filesEndpoint(app, { ...di });
+      handler = app.get.args[0].pop();
```
[the remark this answers is not in the document]
Comment (tbolt) [reply api/routes/auth/certifications/files.test.js]: Updated

**Comment by knollfear** [comment]
|Given|When|Then|Result|
|---|---|---|---|
|Logged in as fed admin|uploads a valid PDF|it returns a URL|:white_check_mark: PASS|
|Logged in as fed admin|uploads an invalid file type|it responds with 415 (invalid file type)|:white_check_mark: PASS|
|Logged in as fed admin|requests the file that was previously uploaded|sends the file|:white_check_mark: PASS|
|Logged in as fed admin|requests a non existent file|responds with 400 Bad Request|:white_check_mark: PASS|
|Logged in as non admin|uploads a valid PDF|it returns 403 (Unauthorized)|:white_check_mark: PASS|
|Logged in as fed admin|uploads an invalid file type|it responds with 403 (Unauthorized)|:white_check_mark: PASS|
|Logged in as fed admin|requests the file that was previously uploaded|it responds with 403 (Unauthorized)|:white_check_mark: PASS|
|Logged in as fed admin|requests a non existent file|it responds with 403 (Unauthorized)|:white_check_mark: PASS|

**Comment by thetif** [comment]
Jerome was saying this week that he doesn't need to verify backend tickets. I think this would count, especially since he will be able to test this functionality in the next ticket that adds the frontend component. So I think you are good to merge this ticket.

**Pull request merged**